backend/src/utils/tools.py

```
@@ -44,17 +64,36 @@ def call_tool_with_interrupt(config: RunnableConfig, **tool_input):
+
+        # update tool call args - SECURITY FIX: validate edited args against schema
         elif response["type"] == "edit":
-            tool_input = response["args"]["args"]
-            tool_response = tool.invoke(tool_input, config)
+            edited_args = response["args"]["args"]
+
+            # Re-validate edited args against the tool's schema before execution
+            if tool.args_schema:
+                try:
+                    validated_args = tool.args_schema(**edited_args).model_dump()
+                except Exception as e:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)